### DISCUSSION TITLE
Is it possible to rescale/reproject tiles?

### DISCUSSION THREAD

**Discussion by matthewtboyle** [Leaflet/Leaflet#9551] [Q&A] [answered]
Hello,

 I use leaflet and several plugins including geoman, but I think here is the place to ask this question. If it isn't, my apologies.

In my project, a user can upload a .dwg file that will be converted into map tiles using 3rd party software that I do not have control over. They can then place markers on various places around their map.

If they make changes to their .dwg file, they can upload a new one and new tiles are created. This is where sometimes the problem will occur.

Depending on the changes made to the .dwg file, the new copy of the map can be offset compared to the original, causing the markers to be misaligned. This is a solvable issue as I have created a future where the user can move all markers in unison to re-align them to the map. 

Where the issue I need help solving CAN occur is when the user adds elements outside of the original border of their map. This will cause the map on zoom level 0 to be slightly more zoomed out than it was originally ( because now there is more that needs to be fit inside of the 256x256 .jpeg ). All of the subsequent zoom levels are affected the same way. Moving markers in unison cannot solve this problem, because the space between the markers has now changed in relation to the map.

What I am looking for is a way to reproject or rescale the map tiles so that they fit the marker positions, without requiring users to manually adjust each marker individually.

Is there a solution in Leaflet or any plugin that can help with this kind of issue?

I hope I made myself clear, please let me know if you have any questions or need clarifications.
Thank you

**Comment by IvanSanchez** [comment, marked as answer]
Short answer: no. Leaflet cannot reproject tile layers.

Long answer:

Your *specific* case is a [GIGO](https://en.wikipedia.org/wiki/Garbage_in,_garbage_out) problem: the tiles move every time the slicing is done, which means the entire dataset is changing coordinates. From Leaflet's point of view, the (0,0) coordinate should always be in one of the corners of the [0,0] tile, and that's not the case. Remember, `.dwg` files do have a coordinate system and a [0,0] coordinate. The 3rd party software *should* care about that.

You *might* get away with using `proj4leaflet`, and defining a custom CRS with different custom scale factors. Do research that, and do research the output of the 3rd party tool. Is there any metadata about the scale factors (`.dwg` units per tile pixel) and the offset relative to the origin of coordinates?

**Reply by matthewtboyle** [reply]
Thank you for such a quick reply. I very much appreciate you pointing me in the right direction. I do not have an answer to your question yet, but it is something that I can find out as I dig. Thanks again.
